.github/workflows/release.yml

```diff
@@ -35,7 +36,7 @@ jobs:
           echo "scope=@mands" > .npmrc
           echo "@mands:registry=https://registry.npmjs.org/" >> .npmrc
           echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" >> .npmrc
-          yarn publish --non-interactive --access=public
+          lerna publish --yes --access=public
```

Comment:
Where does this version end up? Would this commit an update to `lerna.json` when it runs on main?

Reply:
yeah, lerna does the bump itself, using the `lerna.json` package to determine the current version (and bumping the version of anything that has a diff from `main`). It typically also pushes a tag to the repo with a bumped version number

I added the workflow_trigger for the workflow so you can examine for yourself (as you know, ci with packages requires a tiny bit of hands-om and workflows need to be merged before they are available in the trigger menu).

If you go with a different tool cool too, lerna just has the blessing of nx now (because nrwl owns it).

Reply:
if you want to try this, my advice would be to merge it in with the auto-publishing commented, and then try the workflow trigger.  this tends to be how i test autopublishing in a monorepo.

as always, when apikeys are involved, you probably want to verify by hand first.

Reply:
Happy to use lerna but we have branch protection on main so we won't be able to push anything once this runs on main

Reply:
there definitely is a way, though all the ones I know about are in private repos from which I can't directly copy code.  it usually does involve using a bot user for the autopublishing.

there is some discussion of [approaches](https://github.com/lerna/lerna/issues/1957) here, maybe something suits:

Reply:
the only real alternative @andreacaldera is that if publishing is exclusively done via the `workspace_trigger` - rather than automatically - as long as the github user running it has the ability to bypass branch protection it should work that way too.   less automatic but commonly done as well.

just trying to give you guys an option besides the manual-version-number-bumping, which is error prone and a headache for contributors.

Reply:
one final alternative might be to avoid lerna but change your current publish script to
- lint that all versions are consistent
- only run when a `git diff` indicates there was a change to a version in a `package.json

publishing via manual tweaking of versions would still be a little odd, but at least it becomes _your_ problem :)

Reply:
the only real alternative is that if publishing is exclusively done via the `workspace_trigger` - rather than automatically - as long as the github user running it has the ability to bypass branch protection it should work that way too. less automatic but commonly done as well.

just trying to give you guys an option besides the manual-version-number-bumping, which is error prone and a headache for contributors (me at least...).

------------------------------------------------------------------------------------------------------------------

one final alternative might be to avoid `lerna` but change your current publish script to

- lint that all versions are consistent
- only run when a `git diff` indicates there was a change to a version in a `package.json`

publishing via manual tweaking of versions would still be a little odd, but at least it becomes _your_ problem :)

Reply:
there's also a lot of forum love for https://github.com/jscutlery/semver if you wanted something less heavy (though with a smaller community) @andreacaldera .

Reply:
there's also this, but I haven't used it @andreacaldera 

https://github.blog/changelog/2022-08-18-bypass-branch-protections-with-a-new-permission/

Reply:
Thanks a lot of looking into this @ludicast and sorry for the slow response. 
It's on our radar, we'll look into this as soon as we can find a bit of time ;-)